### PR TITLE
Change proofs of implementation

### LINKED ISSUE #112: Change proofs of implementation
The current trick doesn't work as intended

```ts
// Option.ts

// tslint:disable-next-line no-unused-expression
;(
  { map, of, ap, chain, reduce, traverse, zero, alt, extend } as (
    Monad<URI> &
    Foldable<URI> &
    Plus<URI> &
    Traversable<URI> &
    Alternative<URI> &
    Extend<URI>
  )
)
```

`URI` is missing but goes unnoticed. This seems to work better

```ts
const proof:
  Monad<URI> &
  Foldable<URI> &
  Traversable<URI> &
  Alternative<URI> &
  Extend<URI>
= { map, of, ap, chain, reduce, traverse, zero, alt, extend }
// tslint:disable-next-line no-unused-expression
{ proof }
```

```
Error: Property 'URI' is missing in type '{ map: <A, B>(f: (a: A) => B, fa: Option<A>) => Option<B>; of: <A>(a: A) => Option<A>; ap: <A, B>...'.
```

### PR DESCRIPTION
**Pull request by rilut**
closes #112

### CODE REVIEW

File: src/Writer.ts [hunk 1]
```diff
@@ -76,9 +76,8 @@ export function getMonad<W>(monoid: Monoid<W>): Monad<URI> {
   }
 }
 
+const proof:
```
Comment (rilut) [inline src/Writer.ts]: @gcanti should `Writer` have `Monad` proof too?

**Comment by gcanti** [comment]
@rilut FYI you can run tests and linter locally with `npm test`

File: src/Writer.ts (hunk 1 above)
Comment (gcanti) [inline src/Writer.ts]: Theoretically the proof should be provided by the return type of `getMonad`

**Comment by gcanti** [comment]
@rilut Travis is failing because `URI` is missing in a bunch of files, For example in `Option.ts`

```diff
const proof:
  Monad<URI> &
  Foldable<URI> &
  Plus<URI> &
  Traversable<URI> &
  Alternative<URI> &
  Extend<URI>
-= { map, of, ap, chain, reduce, traverse, zero, alt, extend }  
+= { URI, map, of, ap, chain, reduce, traverse, zero, alt, extend }
// tslint:disable-next-line no-unused-expression
{ proof }
```

If you run `npm run build` locally, TypeScript will tell you which files must be amended

**Comment by gcanti** [comment]
Awesome, thanks a lot @rilut. May I merge?

**Comment by rilut** [comment]
@ganti sure!

**Comment by rilut** [comment]
Thanks for your help, @gcanti

**Pull request merged**